### PR TITLE
Ispn 8062/accept or generate password

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-8062

This PR has been opened as a follow up of https://github.com/jboss-dockerfiles/infinispan/pull/47

### CODE REVIEW

File: server/docker-entrypoint.sh [hunk 1]
```diff
@@ -12,12 +13,104 @@ fi
...
+    echo "#         ...                                                                        #"
+    echo "#         env:                                                                       #"
+    echo "#         - name: MGMT_USER                                                          #"
+    echo "#           value: admin                                                             #"
+    echo "#         - name: MGMT_PASS                                                          #"
+    echo "#           value: admin                                                             #"
+    echo "#                                                                                    #"
+    echo "# OpenShift client example:                                                          #"
+    echo "#     oc new-app ... -e MGMT_USER=user -e MGMT_PASS=changeme ...                     #"
+    echo "######################################################################################"
```
Comment (gustavonalle) [inline server/docker-entrypoint.sh]: Could we skip the ```Dockerfile example```, ```Kubernetes Example``` and ```Openshift client example``` from this banner? 
The banner is printed 2x if I don't specify the passowrd and it seemed too verbose for me.

File: server/README.md [hunk 2]
```diff
@@ -42,14 +96,6 @@ The first param to the container is the name of the desired configuration. For e
...
-   
-The 'default' and 'standalone' running modes don't not have credentials set. In order to define them, run after launching the container:
-
-    docker exec -it $(docker ps -l -q) /opt/jboss/infinispan-server/bin/add-user.sh
-
-and follow the instructions.
-
 ## Running domain mode
 
 Domain mode is composed of a lightweight managing process that does not hold data called domain controller plus one or more
```
Comment (gustavonalle) [inline server/README.md]: Running domain mode without setting a password now gives an exception:

```
[Host Controller] Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[12,21]
[Host Controller] Message: "WFLYCTL0113: '' is an invalid value for parameter value. Values must have a minimum length of 1 characters"
[Host Controller]       at org.jboss.as.controller.SimpleAttributeDefinition.parse(SimpleAttributeDefinition.java:161)
[Host Controller]       at org.jboss.as.controller.SimpleAttributeDefinition.parseAndSetParameter(SimpleAttributeDefinition.java:186)
[Host Controller]       at org.jboss.as.domain.management.parsing.ManagementXml_4.parseSecret(ManagementXml_4.java:528)
[Host Controller]       at org.jboss.as.domain.management.parsing.ManagementXml_4.parseServerIdentities(ManagementXml_4.java:503)
[Host Controller]       at org.jboss.as.domain.management.parsing.ManagementXml_4.parseSecurityRealm(ManagementXml_4.java:453)
[Host Controller]       at org.jboss.as.domain.management.parsing.ManagementXml_4.parseSecurityRealms(ManagementXml_4.java:421)
[Host Controller]       at org.jboss.as.domain.management.parsing.ManagementXml_4.parseManagement(ManagementXml_4.java:176)
```

We should:

1) Make mandatory setting ```-e "MGMT_USER=...." -e "MGMT_PASS=..."``` when using domain mode (and don't use auto-generation, nor print the banner)
2) Mention in this section that both the domain-controller AND the host-controller must be started with the MGMT_USER and MGMT_PASS variables

File: server/README.md [hunk 3]
```diff
@@ -1,5 +1,59 @@
 # Infinispan server Docker image
 
+## Configuring authentication
+
+To be able to connect to any of the Infinispan server Docker images, authentication is necessary.
+The easiest way to create a new user (with specified password) before starting the server is to specify `APP_USER`
+and `APP_PASS` environment variables or pass `-au` (for user name) and `-au` (for password) switches.
```
Comment (slaskawi) [inline server/README.md]: Done

File: server/docker-entrypoint.sh (hunk 1 above)
Comment (slaskawi) [inline server/docker-entrypoint.sh]: I actually left it as is. I think it has some value. Besides we can remove it whenever we want without breaking any backwards compatibility. 

Let's also ask original author, @galderz, what to do with that.

**Comment by gustavonalle** [comment]
Integrated, thanks @slaskawi

File: server/docker-entrypoint.sh (hunk 1 above)
Comment (galderz) [inline server/docker-entrypoint.sh]: I'm fine either way. I thought it could be useful, but happy to remove it if its too noisy/big.

**Comment by galderz** [comment]
Nice stuff @slaskawi, thanks! :)

**Pull request merged**